mlserver/pydantic_migration.py

```diff
@@ -0,0 +1,19 @@
+# noqa: F401
+from typing import TYPE_CHECKING
+from pydantic import VERSION
+
+
+def is_pydantic_v1():
+    return VERSION.startswith("1.")
+
+
+if TYPE_CHECKING:
+    from pydantic.types import ImportString as PyObject
+    from pydantic_settings import BaseSettings
+else:
+    if is_pydantic_v1():
+        from pydantic import PyObject
+        from pydantic import BaseSettings
+    else:
+        from pydantic.types import ImportString as PyObject
+        from pydantic_settings import BaseSettings
```

Comment:
This seems ugly, but it seems to be the only way to trick MyPy (link to a related issue [here](https://github.com/pydantic/pydantic/issues/6022#issuecomment-1671630257)).

How this goes:
1) If `TYPE_CHECKING`, import the objects from the pydantic version installed in the environment. As this is the import that runs first, `mypy` will use these imported objects when linting.
2) `else` imports as used during runtime.